rfcs/models/security/confidentiality-design-choices.md

```diff
@@ -0,0 +1,174 @@
+<!--
+ Copyright IBM Corp. All Rights Reserved.
+
+ SPDX-License-Identifier: CC-BY-4.0
+ -->
+# Design Choices for End-to-End Confidentiality Mechanism
+
+- RFC: 01-008-appendix
+- Authors: Venkatraman Ramakrishna, Dhinakaran Vinayagamurthy, Sandeep Nishad, Krishnasuri Narayanam, Sikhar Patranabis
+- Status: Proposed
+- Since: 10-May-2022
+
+
+## Summary
+
+To fulfil the requirements described in the [end-to-end confidentiality spec](./confidentiality.md), the view contents returned by the source ledger must contain some unencrypted portions in the signed payload that can validate the plaintext contents as decrypted by the destination client. This allows the destination peers to validate the plaintext against tampering by the client. The solution for this was to add a hash of some form within the interoperation payload in the view.
+
+## General Considerations
+
+For private (confidential) portions of a view's contents:
+- They should be encrypted with the receiver's public key.
+- If they only need tamper protection, they should be hashed.
+- If they need tamper protection and need to be authenticated by the receiver, they should be hashed and signed.
+For public (non-confidential) portions of a view's contents:
+- If they need to be authenticated by the receiver, they should be signed.
+- If they only need tamper protection, they can be either hashed or signed.
+(_Note_: "hashed" implies "HMAC")
```

Comment:
You might want to say that we use HMAC for provable security guarantees. Or you could skip the HMAC reference altogether here.

Reply:
Removed. I don't know why I added it in the first place.